src/party.cpp

```diff
@@ -419,7 +421,7 @@ void Party::shareExperience(uint64_t experience, Creature* source/* = nullptr*/)
 	leader->onGainSharedExperience(shareExperience, source);
 }
 
-bool Party::canUseSharedExperience(const Player* player) const
+bool Party::canUseSharedExperience(const Player* player)
```

Comment:
sorry, i didnt read the entire function canUseSharedExperience, there are more "exits" from it than I thought
maybe you can use return values like other queries and create enum for it